### ISSUE TITLE
Resizing existing terrain

### ISSUE THREAD

**Issue by Wallace99** [Zylann/godot_heightmap_plugin#136]
How should resizing with the stretch option work? I assumed from the docs that the entire terrain would be scaled up from 513 - 1025 but what I seem to get instead is just my 513 sized terrain surrounded by a flat black plane.

In the console there are a tonne of these:

>  core/image.cpp:2395 - Index p_x = 955 is out of bounds (width = 513).
 core/image.cpp:2395 - Index p_x = 956 is out of bounds (width = 513).
 core/image.cpp:2395 - Index p_x = 957 is out of bounds (width = 513).
 core/image.cpp:2395 - Index p_x = 958 is out of bounds (width = 513).
 core/image.cpp:2395 - Index p_x = 959 is out of bounds (width = 513).
 core/image.cpp:2395 - Index p_x = 960 is out of bounds (width = 513).
 core/image.cpp:2395 - Index p_x = 944 is out of bounds (width = 513).
 core/image.cpp:2395 - Index p_x = 945 is out of bounds (width = 513).
 core/image.cpp:2395 - Index p_x = 946 is out of bounds (width = 513).
 core/image.cpp:2395 - Index p_x = 947 is out of bounds (width = 513).
 core/image.cpp:2395 - Index p_x = 948 is out of bounds (width = 513).
 core/image.cpp:2395 - Index p_x = 949 is out of bounds (width = 513).
 core/image.cpp:2395 - Index p_x = 950 is out of bounds (width = 513).
 core/image.cpp:2395 - Index p_x = 951 is out of bounds (width = 513).
 core/image.cpp:2395 - Index p_x = 952 is out of bounds (width = 513).
 core/image.cpp:2395 - Index p_x = 953 is out of bounds (width = 513).
 core/image.cpp:2395 - Index p_x = 954 is out of bounds (width = 513).
 core/image.cpp:2395 - Index p_x = 955 is out of bounds (width = 513).
 core/image.cpp:2395 - Index p_x = 956 is out of bounds (width = 513).
 core/image.cpp:2395 - Index p_x = 957 is out of bounds (width = 513).
 core/image.cpp:2395 - Index p_x = 958 is out of bounds (width = 513).
 core/image.cpp:2395 - Index p_x = 959 is out of bounds (width = 513).
 core/image.cpp:2395 - Index p_x = 960 is out of bounds (width = 513).
 core/image.cpp:2395 - Index p_x = 944 is out of bounds (width = 513).
 core/image.cpp:2395 - Index p_x = 945 is out of bounds (width = 513).
 core/image.cpp:2395 - Index p_x = 946 is out of bounds (width = 513).
 core/image.cpp:2395 - Index p_x = 947 is out of bounds (width = 513).
 core/image.cpp:2395 - Index p_x = 948 is out of bounds (width = 513).
 core/image.cpp:2395 - Index p_x = 949 is out of bounds (width = 513).
 core/image.cpp:2395 - Index p_x = 950 is out of bounds (width = 513).
 core/image.cpp:2395 - Index p_x = 951 is out of bounds (width = 513).
 core/image.cpp:2395 - Index p_x = 952 is out of bounds (width = 513).
 core/image.cpp:2395 - Index p_x = 953 is out of bounds (width = 513).
 core/image.cpp:2395 - Index p_x = 954 is out of bounds (width = 513).
 core/image.cpp:2395 - Index p_x = 955 is out of bounds (width = 513).
 core/image.cpp:2395 - Index p_x = 956 is out of bounds (width = 513).
 core/image.cpp:2395 - Index p_x = 957 is out of bounds (width = 513).
 core/image.cpp:2395 - Index p_x = 958 is out of bounds (width = 513).
 core/image.cpp:2395 - Index p_x = 959 is out of bounds (width = 513).
 core/image.cpp:2395 - Index p_x = 960 is out of bounds (width = 513).
 core/image.cpp:2395 - Index p_x = 960 is out of bounds (width = 513).
 core/image.cpp:2395 - Index p_x = 960 is out of bounds (width = 513).
 core/image.cpp:2395 - Index p_x = 961 is out of bounds (width = 513).
 core/image.cpp:2395 - Index p_x = 962 is out of bounds (width = 513).
 core/image.cpp:2395 - Index p_x = 963 is out of bounds (width = 513).
 core/image.cpp:2395 - Index p_x = 964 is out of bounds (width = 513).
 core/image.cpp:2395 - Index p_x = 965 is out of bounds (width = 513).
 core/image.cpp:2395 - Index p_x = 966 is out of bounds (width = 513).
 core/image.cpp:2395 - Index p_x = 967 is out of bounds (width = 513).
 core/image.cpp:2395 - Index p_x = 968 is out of bounds (width = 513).
 core/image.cpp:2395 - Index p_x = 969 is out of bounds (width = 513).
 core/image.cpp:2395 - Index p_x = 970 is out of bounds (width = 513).
 core/image.cpp:2395 - Index p_x = 971 is out of bounds (width = 513).
 core/image.cpp:2395 - Index p_x = 972 is out of bounds (width = 513).
 core/image.cpp:2395 - Index p_x = 973 is out of bounds (width = 513).
 core/image.cpp:2395 - Index p_x = 974 is out of bounds (width = 513).
 core/image.cpp:2395 - Index p_x = 975 is out of bounds (width = 513).
 core/image.cpp:2395 - Index p_x = 976 is out of bounds (width = 513).
 core/image.cpp:2395 - Index p_x = 960 is out of bounds (width = 513).
 core/image.cpp:2395 - Index p_x = 961 is out of bounds (width = 513).
 core/image.cpp:2395 - Index p_x = 962 is out of bounds (width = 513).

I'm on MacOS using godot 3.2

**Comment by Zylann** [comment]
The terrain will be surrounded by flat planes if stretching is turned off, alrhough there seems to be an issue here

**Comment by Wallace99** [comment]
I can't think of anything I've done differently that could have caused it to act differently. I had experimented a bit in script with iterating over the heightmap and looking at the height values as I was going to try write a script to place trees in forests. I noticed in particular in the error logs something about the image being locked so thought back to when I'd locked the image in the script but I assumed that wouldn't persist? I've got my project in a git repo if that would help:

https://github.com/Wallace99/godot-project

> HeightMapData::set_resolution 1025
Resizing height[0]...
 Cannot resize image when it is locked.
Resizing normal[0]...
Resizing splat[0]...
Resizing color[0]...
Resizing detail[0]...
Resizing detail[1]...
 Cannot resize image before creating it, use create() or create_from_data() first.
Resizing global_albedo[0]...
Updating all vertical bounds... (64x64 chunks)
 core/image.cpp:2395 - Index p_x = 513 is out of bounds (width = 513).
 core/image.cpp:2395 - Index p_x = 514 is out of bounds (width = 513).
 core/image.cpp:2395 - Index p_x = 515 is out of bounds (width = 513).
 core/image.cpp:2395 - Index p_x = 516 is out of bounds (width = 513).
 core/image.cpp:2395 - Index p_x = 517 is out of bounds (width = 513).
 core/image.cpp:2395 - Index p_x = 518 is out of bounds (width = 513).
 core/image.cpp:2395 - Index p_x = 519 is out of bounds (width = 513).
 core/image.cpp:2395 - Index p_x = 520 is out of bounds (width = 513).
 core/image.cpp:2395 - Index p_x = 521 is out of bounds (width = 513).
 core/image.cpp:2395 - Index p_x = 522 is out of bounds (width = 513).
 core/image.cpp:2395 - Index p_x = 523 is out of bounds (width = 513).
 core/image.cpp:2395 - Index p_x = 524 is out of bounds (width = 513).
 core/image.cpp:2395 - Index p_x = 525 is out of bounds (width = 513).
 core/image.cpp:2395 - Index p_x = 526 is out of bounds (width = 513).
 core/image.cpp:2395 - Index p_x = 527 is out of bounds (width = 513).
 core/image.cpp:2395 - Index p_x = 528 is out of bounds (width = 513).
 core/image.cpp:2395 - Index p_x = 513 is out of bounds (width = 513).
 core/image.cpp:2395 - Index p_x = 514 is out of bounds (width = 513).
 core/image.cpp:2395 - Index p_x = 515 is out of bounds (width = 513).
 core/image.cpp:2395 - Index p_x = 516 is out of bounds (width = 513).
 core/image.cpp:2395 - Index p_x = 517 is out of bounds (width = 513).
 core/image.cpp:2395 - Index p_x = 518 is out of bounds (width = 513).
 core/image.cpp:2395 - Index p_x = 519 is out of bounds (width = 513).
 core/image.cpp:2395 - Index p_x = 520 is out of bounds (width = 513).
 core/image.cpp:2395 - Index p_x = 521 is out of bounds (width = 513).

Before (513 resolution):
https://i.ibb.co/3f9hnxC/Screen-Shot-2020-04-03-at-9-47-00-AM.png

After (1025 resolution - although I notice none of the albedo, detail pngs have been resized)
https://i.ibb.co/k5c8645/Screen-Shot-2020-04-03-at-9-49-12-AM.png

**Comment by Wallace99** [comment]
Just tried now with a blank project and same result. I created a hterrain with a folder called terrain_data selected as the resource folder. I then generate using the default settings -> apply. Then resize from 513 to 1025 and the same error logs happened with the terrain being stretched in x and y but flat in the far corner as shown in the screenshots previously ^ Only other thing I can think of is that I changed the shaderType from classic4 to classic4Lite because with classic4 my terrain is entirely black and this seemed to be a work around
